vm-virtio/src/console.rs

```diff
@@ -331,12 +398,33 @@ impl VirtioDevice for Console {
         self.acked_features |= v;
     }
 
-    fn read_config(&self, _offset: u64, _data: &mut [u8]) {
-        warn!("Device specific configuration is not defined yet");
+    fn read_config(&self, offset: u64, mut data: &mut [u8]) {
+        let config = self.config.lock().unwrap();
+        let config_slice = config.as_slice();
+        let config_len = config_slice.len() as u64;
+        if offset >= config_len {
+            error!("Failed to read config space");
+            return;
+        }
+
+        if let Some(end) = offset.checked_add(data.len() as u64) {
+            // This write can't fail, offset and end are checked against config_len.
+            data.write_all(&config_slice[offset as usize..cmp::min(end, config_len) as usize])
+                .unwrap();
+        }
     }
 
-    fn write_config(&mut self, _offset: u64, _data: &[u8]) {
-        warn!("Device specific configuration is not defined yet");
+    fn write_config(&mut self, offset: u64, data: &[u8]) {
```

Comment:
Looking at the spec I think the fields are completely read-only from the driver site.

Reply:
From the the console size perspective, Yes. However, for console size we have exposed virtio_console_config struct, where emerg_wr is a part of it. In future, we plan to enable emerg_wr for which we may need write_config. Therefore, I kept it implemented.

Reply:
You should delete it for now as we aren't advertising that feature (and i'm not convinced it's a very useful one for us). Carrying some code that is unused isn't helpful to us.

Reply:
Ok.